sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -231,19 +259,101 @@ public SamplingResult shouldSample(
         Kind spanKind,
         ReadableAttributes attributes,
         List<Link> parentLinks) {
-      if (parentContext.isValid()) {
-        if (parentContext.getTraceFlags().isSampled()) {
-          return EMPTY_RECORDED_AND_SAMPLED_SAMPLING_RESULT;
-        }
-        return EMPTY_NOT_SAMPLED_OR_RECORDED_SAMPLING_RESULT;
+      if (!parentContext.isValid()) {
+        return this.root.shouldSample(
+            parentContext, traceId, name, spanKind, attributes, parentLinks);
       }
-      return this.delegateSampler.shouldSample(
-          parentContext, traceId, name, spanKind, attributes, parentLinks);
+
+      if (parentContext.isRemote()) {
+        return parentContext.getTraceFlags().isSampled()
+            ? this.remoteParentSampled.shouldSample(
+                parentContext, traceId, name, spanKind, attributes, parentLinks)
+            : this.remoteParentNotSampled.shouldSample(
+                parentContext, traceId, name, spanKind, attributes, parentLinks);
+      }
+      return parentContext.getTraceFlags().isSampled()
+          ? this.localParentSampled.shouldSample(
+              parentContext, traceId, name, spanKind, attributes, parentLinks)
+          : this.localParentNotSampled.shouldSample(
+              parentContext, traceId, name, spanKind, attributes, parentLinks);
     }
 
     @Override
     public String getDescription() {
-      return String.format("ParentOrElse{%s}", this.delegateSampler.getDescription());
+      return String.format(
+          "ParentBased{root:%s,remoteParentSampled:%s,remoteParentNotSampled:%s,"
+              + "localParentSampled:%s,localParentNotSampled:%s}",
+          this.root.getDescription(),
+          this.remoteParentSampled.getDescription(),
+          this.remoteParentNotSampled.getDescription(),
+          this.localParentSampled.getDescription(),
+          this.localParentNotSampled.getDescription());
+    }
+
+    static class Builder {
+      private final Sampler root;
+      private Sampler remoteParentSampled;
+      private Sampler remoteParentNotSampled;
+      private Sampler localParentSampled;
+      private Sampler localParentNotSampled;
+
+      /**
+       * Sets the {@link Sampler} to use when there is a remote parent that was sampled.
+       *
+       * @return this Builder
+       */
+      public Builder withRemoteParentSampled(Sampler remoteParentSampled) {
+        this.remoteParentSampled = remoteParentSampled;
+        return this;
+      }
+
+      /**
+       * Sets the {@link Sampler} to use when there is a remote parent that was not sampled.
```

Comment:
If not set, defaults to never sampling when the remote parent isn't sampled.

Similar for local